src/content/en/showcase/2017/with.in.md

```diff
@@ -0,0 +1,331 @@
+
+Extruded 3D text is essentially the same as triangulated vector text, but with depth and possibly a bevel, so it has at least twice as much geometry. Either of these could work in small doses for titles or logos, but wouldn't perform as well for large quantities of text, and neither has typographic features.
+
+Bitmap fonts use one quad (two triangles) per character, so it uses less geometry and performs better than [triangulated vectors](https://vr.with.in/archive/text-triangulated-vector/). They're still raster based since they use a texture map sprite, but with an SDF shader, they're basically resolution independent, so they look nicer than a 2D canvas texture. Matt DesLauriers' three-bmfont-text also includes reliable typographic features for text wrapping, letter spacing, line height and alignment. Overflow doesn't get cut off. Font size is controlled through scale. We chose this route because it gave us the best options for design while still staying performant. Unfortunately, it wasn't as easy to implement so we'll go through the steps in hopes to alleviate fellow developers working in WebVR.
+
+[mattdesl.svbtle.com/material-design-on-the-gpu](https://mattdesl.svbtle.com/material-design-on-the-gpu)
+
+[mattdesl.github.io/physical-text/index.html](http://mattdesl.github.io/physical-text/index.html)
+
+[github.com/mattdesl/physical-text](https://github.com/mattdesl/physical-text)
+
+
+<figure>
+  <img src="/web/showcase/2017/images/with.in/diagram.png">
+  <figcaption>Font to SDF Bitmap Workflow</figcaption>
+</figure>
+
+
+## 1. Generate bitmap font (.png + .fnt)
+
+[libgdx.badlogicgames.com/tools.html](https://libgdx.badlogicgames.com/tools.html)
+
+[github.com/libgdx/libgdx/wiki/Hiero](https://github.com/libgdx/libgdx/wiki/Hiero)
+
+[github.com/libgdx/libgdx/wiki/Distance-field-fonts](https://github.com/libgdx/libgdx/wiki/Distance-field-fonts)
+
+Hiero is a bitmap font packing tool that runs with Java. The Hiero documentation doesn't really explain how to run it without going through a complicated build process. First, install Java if you haven't already. Then if double clicking on the runnable-hiero.jar doesn't open Hiero, try running it with this command in the console:
+
+
+```
+java -jar runnable-hiero.jar
+```
+
+
+Once Hiero is running, open a .ttf or .otf desktop font, enter any extra characters you want included, change Rendering to Java to enable Effects, increase the Size so that your characters fill up the entire Glyph cache square, add a Distance Field Effect, adjust the Distance field's scale and spread. The scale value is like a resolution. The higher it is, the less blurry it will be, but the longer it will take for Hiero to render the preview. Then save your bitmap font. It generates a bitmap font is consisting of a .png image and an AngelCode .fnt font description file.
+
+
+
+<figure>
```

Comment:
Done. Actually ended up floating a few of these, because it looked weird doing just one.